src/components/preferences/Preferences.js

```diff
@@ -0,0 +1,463 @@
+import { injectIntl } from "react-intl";
+
+import ids from "./ids";
+import messages from "./messages";
+import constants from "../../constants";
+import prefConstants from "./constants";
+import General from "./General";
+import Shortcuts from "./Shortcuts";
+import styles from "./styles";
+
+import { isWritable } from "../data/utils";
+import { useUserProfile } from "../../contexts/userProfile";
+
+import {
+    bootstrap,
+    getRedirectURIs,
+    savePreferences,
+    resetToken,
+    BOOTSTRAP_KEY,
+} from "../../serviceFacades/users";
+import { getResourceDetails } from "../../serviceFacades/filesystem";
+import GridLoading from "../utils/GridLoading";
+import withErrorAnnouncer from "../utils/error/withErrorAnnouncer";
+
+import {
+    announce,
+    AnnouncerConstants,
+    build,
+    getMessage,
+    formatMessage,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+
+import {
+    Button,
+    CircularProgress,
+    Container,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    Divider,
+    Grid,
+    Paper,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(styles);
+
+function Preferences(props) {
+    const { baseId, intl, showErrorAnnouncer } = props;
+    const [userProfile] = useUserProfile();
+    const [showRestoreConfirmation, setShowRestoreConfirmation] = useState(
+        false
+    );
+    const [userPref, setUserPref] = useState({});
+    const [fetchDetailsKey, setFetchDetailsKey] = useState("");
+    const [defaultOutputFolder, setDefaultOutputFolder] = useState(null);
+    const [requireAgaveAuth, setRequireAgaveAuth] = useState(true);
+    const [
+        outputFolderValidationError,
+        setOutputFolderValidationError,
+    ] = useState(null);
+    const [bootstrapKey, setBootstrapKey] = useState("");
+    const [fetchRedirectURIsKey, setFetchRedirectURIsKey] = useState("");
+    const [hpcAuthUrl, setHPCAuthUrl] = useState("");
+
+    const classes = useStyles();
+
+    const preProcessData = (respData) => {
+        let pref = respData.preferences;
+        setDefaultOutputFolder(
+            pref?.default_output_folder?.path ||
+                pref?.system_default_output_dir?.path
+        );
+        setBootstrapKey(null);
+        setUserPref(pref);
+        const session = respData?.session;
+        const agaveKey = session?.auth_redirect?.agave;
+        if (agaveKey) {
+            setRequireAgaveAuth(true);
+        } else {
+            setRequireAgaveAuth(false);
+        }
+    };
+
+    useEffect(() => {
+        //get from cache if not fetch now.
+        const ip = userProfile?.attributes.ip;
+        const prefCache = queryCache.getQueryData([BOOTSTRAP_KEY, { ip }]);
+        if (prefCache) {
+            preProcessData(prefCache);
+        } else {
+            setBootstrapKey([BOOTSTRAP_KEY, { ip }]);
+        }
+    }, [userPref, userProfile]);
```

Comment:
Fetch back cached bootstrap info. If not, call bootstrap service again and cache the result.